merino/providers/weather/backends/accuweather.py

```diff
@@ -92,6 +110,87 @@ def cache_inputs_for_weather_report(self, geolocation: Location) -> Optional[byt
 
         return (geolocation.country + geolocation.postal_code).encode("utf-8")
 
+    def cache_key_for_accuweather_request(
+        self, url: str, query_params: dict[str, str] = {}
+    ) -> str:
+        """Get the cache key for the accuweather request.
+        Also ensure that the API key is stripped out.
+        """
+        params = query_params.copy()
+        if "apikey" in params:
+            del params["apikey"]
+
+        if params:
+            extra_identifiers = hashlib.blake2s(str(params).encode("utf-8")).hexdigest()
+            return f"{self.__class__.__name__}:v1:{url}:{extra_identifiers}"
+        return f"{self.__class__.__name__}:v1:{url}"
+
+    async def get_request(
+        self, client: AsyncClient, url: str, params: dict[str, str] = {}
+    ) -> dict[str, Any]:
+        """Get API response. Attempt to get it from cache first,
+        then actually make the call if there's a cache miss.
+        """
+        cache_key = self.cache_key_for_accuweather_request(url, params)
+        response_dict: dict[str, str]
+
+        # The top level path in the URL gives us a good enough idea of what type of request
+        # we are calling from here.
+        request_type: str = url.strip("/").split("/")[0]
```

Comment:
Since we only care about the 1st entry, no need to split the rest.
```suggestion
        request_type: str = url.strip("/").split("/", 1)[0]
```